databricks/koalas/frame.py

```diff
@@ -2696,8 +2696,8 @@ def rename(index):
             # Now, new internal Spark columns are named as same as index name.
             new_index_map = [(column, name) for column, name in new_index_map]
 
-            sdf, index_column = _InternalFrame.attach_default_index(sdf)
-            index_map = [(index_column, None)]
+            sdf = _InternalFrame.attach_default_index(sdf)
```

Comment:
Actually I'd prefer to let the function generate the column name.

Reply:
I think it's better to have the same default index name, rather than a name that can be potentially different. If we worry about the name `__index_level_0__` being conflicted, seems fine here:

```python
>>> ks.range(10).reset_index().explain(True)
```

```
...
== Physical Plan ==
*(4) Project [(_we0#21 - 1) AS __index_level_0__#19, index#14, id#0L]
+- Window [row_number() windowspecdefinition(_w0#20L ASC NULLS FIRST, specifiedwindowframe(RowFrame, unboundedpreceding$(), currentrow$())) AS _we0#21], [_w0#20L ASC NULLS FIRST]
   +- *(3) Sort [_w0#20L ASC NULLS FIRST], false, 0
      +- *(3) Project [(_we0#5 - 1) AS index#14, id#0L, monotonically_increasing_id() AS _w0#20L]
         +- Window [row_number() windowspecdefinition(_w0#4L ASC NULLS FIRST, specifiedwindowframe(RowFrame, unboundedpreceding$(), currentrow$())) AS _we0#5], [_w0#4L ASC NULLS FIRST]
            +- *(2) Sort [_w0#4L ASC NULLS FIRST], false, 0
               +- Exchange SinglePartition, true, [id=#32]
                  +- *(1) Project [id#0L, monotonically_increasing_id() AS _w0#4L]
                     +- *(1) Range (0, 10, step=1, splits=12)
```